src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -1150,14 +1189,29 @@ internal void DefineEnum()
                             ParserStrings.MemberAlreadyDefined,
                             enumerator.Name);
                     }
-                    else
+                    else if (value != null)
                     {
+                        value = Convert.ChangeType(value, underlyingType);
                         definedEnumerators.Add(enumerator.Name);
                         enumBuilder.DefineLiteral(enumerator.Name, value);
-                        if (value < int.MaxValue)
+                    }
+
+                    if (value != 0 && _enumDefinitionAst.Attributes.Any(attr => attr.TypeName.GetReflectionType() == typeof(FlagsAttribute)))
+                    {
+                        if (value < maxValue / 2)
```

Comment:
Shouldn't this be value <= maxValue / 2? 🙂 (or is there a catch I'm unaware of where you can't have an enum value that's equal to the MaxValue of the underlying type?)

Reply:
You're right, it should. Just turns out that that doesn't matter here, because the highest cardinal number you can fit into any integral type is never even :)

```powershell
@([byte],[sbyte],[uint16],[int16],[int],[uint32],[long],[uint64]).ForEach({$_::MaxValue})
255
127
65535
32767
2147483647
4294967295
9223372036854775807
18446744073709551615
```